src/pages/home/sidebar/ProfileAvatarWithIndicator.tsx

```diff
@@ -1,35 +1,31 @@
 import React from 'react';
 import {View} from 'react-native';
-import type {OnyxEntry} from 'react-native-onyx';
-import {withOnyx} from 'react-native-onyx';
+import {useOnyx} from 'react-native-onyx';
 import AvatarWithIndicator from '@components/AvatarWithIndicator';
 import OfflineWithFeedback from '@components/OfflineWithFeedback';
 import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
 import useThemeStyles from '@hooks/useThemeStyles';
 import * as UserUtils from '@libs/UserUtils';
 import ONYXKEYS from '@src/ONYXKEYS';
 
-type ProfileAvatarWithIndicatorOnyxProps = {
-    /** Indicates whether the app is loading initial data */
-    isLoading: OnyxEntry<boolean>;
-};
-
-type ProfileAvatarWithIndicatorProps = ProfileAvatarWithIndicatorOnyxProps & {
+type ProfileAvatarWithIndicatorProps = {
     /** Whether the avatar is selected */
     isSelected?: boolean;
 };
 
-function ProfileAvatarWithIndicator({isLoading = true, isSelected = false}: ProfileAvatarWithIndicatorProps) {
+function ProfileAvatarWithIndicator({isSelected = false}: ProfileAvatarWithIndicatorProps) {
     const styles = useThemeStyles();
     const currentUserPersonalDetails = useCurrentUserPersonalDetails();
+    const [isLoadingOnyxValue] = useOnyx(ONYXKEYS.IS_LOADING_APP);
+    const isLoading = isLoadingOnyxValue ?? true;
```

Comment:
Note: This piece of code mimics the old behavior (`withOnyx` + default value assigment to `isLoading` prop).